virtManager/details/snapshots.py

```diff
@@ -342,6 +349,17 @@ def _create_new_snapshot(self):
                     self.topwin)
         progWin.run()
 
+    def _get_current_mode(self):
+        current = self.vm.get_current_snapshot()
+
+        if current is None:
+            return None
+
+        if current.is_external():
+            return "external"
+
+        return "internal"
```

Comment:
The fact that the mode is returned as a verbatim string as opposed to something like an enum or even just a string constant is a bit icky. I know it's pre-existing, but perhaps you could consider changing that, possibly in a follow-up PR.

Reply:
Agreed that it's icky, I'll leave it to follow-up PR.